src/merkle_tree.rs

```diff
@@ -207,4 +270,24 @@ mod test {
             tree.get_root()
         );
     }
+
+    #[test]
+    fn verify_generated_proof() {
+        let tree = MerkleTree::from_leaves(&["1", "2", "3"]);
+        let Some((proof, idx)) = tree.generate_proof(&"1") else {
+            panic!()
+        };
+        assert!(tree.verify_proof(&"1", &proof, idx))
+    }
+
+    #[test]
+    fn verify_proof_after_tree_modification() {
+        let mut tree = MerkleTree::from_leaves(&["1", "2", "3"]);
+        let Some((proof, idx)) = tree.generate_proof(&"1") else {
+            panic!()
+        };
+        assert!(tree.verify_proof(&"1", &proof, idx));
+        tree.add_element(&"4");
+        assert!(!tree.verify_proof(&"1", &proof, idx));
+    }
```

Comment:
I would maybe add that you can now generate a new proof for 1 and that it's again valid